backend/handlers.go

```diff
@@ -124,3 +125,103 @@ func (cs *server) loginHandler(w http.ResponseWriter, r *http.Request, _ httprou
 		return
 	}
 }
+
+// query handlers
+
+// getQueriesHandler returns a list of queries belonging to ther user.
+func (cs *server) getQueriesHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	userID := r.Context().Value(userID("userID")).(string)
+	queries, err := cs.queryRepo.FindByOwnerId(userID)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	if err = json.NewEncoder(w).Encode(queries); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+// createQueryHandler creates a new query.
+func (cs *server) createQueryHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	userID := r.Context().Value(userID("userID")).(string)
+	decoder := json.NewDecoder(r.Body)
```

Comment:
If you read from `r.Body` you should also do a `defer r.Body.Close()`

Reply:
done